### PR TITLE
chore: Training flow revamp adjustments

### PR DESCRIPTION
**Pull request by dwesolow**
## 📝 Description

This PR adds a few training flow adjustments:
- based on feedback from Max, only learning rate should have step `1e-6`, the rest can stay as is;
- there was a typo in tiling;
- add a link to the documentation for original model (link redirects to the documentation of the advanced settings where original model is described) -> the link will not work because documentation was not released yet.

image
image

## ✨ Type of Change

Select the type of change your PR introduces:

- [X] 🔨 **Refactor** – Non-breaking change which refactors the code base

## 🧪 Testing Scenarios

Describe how the changes were tested and how reviewers can test them too:

- [ ] ✅ Tested manually
- [ ] 🤖 Run automated end-to-end tests

## ✅ Checklist

Before submitting the PR, ensure the following:

- [ ] 🔍 PR title is clear and meaningful
- [ ] ✍️ PR description clearly explains the changes and their reason
- [ ] 📝 I have linked the PR to the corresponding GitHub Issues, if any
- [ ] 💬 I have commented my code, especially in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [ ] ✅ I have added tests that prove my fix is effective or my feature works

### CODE REVIEW

File: web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/training/learning-parameters/learning-parameters-list.component.tsx [hunk 1]
```diff
@@ -58,6 +64,45 @@ const SingleLearningParameter = ({
...
+        };
+
+        const handleLearningRateReset = () => {
+            handleChange({
+                ...parameter,
+                value: parameter.defaultValue,
+            });
+        };
+
+        const learningRateStep = 1e-6;
```
Comment (camiloHimura) [inline web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/training/learning-parameters/learning-parameters-list.component.tsx]: Export this variable and reuse it in `learning-parameters.test.tsx`, so both files remain in sync.

File: web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/training/fine-tune-parameters.component.tsx [hunk 2]
```diff
@@ -57,17 +63,22 @@ export const FineTuneParameters: FC<FineTuneParametersProps> = ({
...
+                    </Flex>
                 </RadioGroup>
 
                 <Flex gap={'size-100'} alignItems={'center'} marginTop={'size-100'}>
                     <Checkbox
                         isEmphasized
                         isSelected={isReshufflingSubsetsEnabled}
                         onChange={onReshufflingSubsetsEnabledChange}
-                        UNSAFE_className={classes.trainModelCheckbox}
+                        UNSAFE_className={styles.trainModelCheckbox}
```
Comment (jpggvilaca) [inline web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/training/fine-tune-parameters.component.tsx:82]: Doesnt make a different but why this change?

Comment (dwesolow) [reply web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/training/fine-tune-parameters.component.tsx:82]: Not to rely on the legacy file.

**Review by jpggvilaca — DISMISSED** [review]

File: web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/training/learning-parameters/learning-parameters-list.component.tsx (hunk 1 above)
Comment (dwesolow) [reply web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/training/learning-parameters/learning-parameters-list.component.tsx]: Files remain in sync even without sharing the same variable because test validates it but I'm fine with extracting it. Updated.

File: web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/training/fine-tune-parameters.component.tsx [hunk 3]
```diff
@@ -29,6 +31,10 @@ export const FineTuneParameters: FC<FineTuneParametersProps> = ({
     isReshufflingSubsetsEnabled,
     onReshufflingSubsetsEnabledChange,
 }) => {
+    const docsUrl = useDocsUrl();
+    // eslint-disable-next-line max-len
+    const originalModelUrl = `${docsUrl}docs/user-guide/geti-fundamentals/model-training-and-optimization/#advanced-settings`;
```
[the remark this answers is not in the document]
Comment (dwesolow) [reply web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/training/fine-tune-parameters.component.tsx:36]: AFAIK we don't have a number with more than 6 fraction digits in configuration.

File: web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/number-parameter-field.component.tsx [hunk 4]
```diff
@@ -24,12 +47,12 @@ export const NumberParameterField: FC<NumberGroupParamsProps> = ({
...
     name,
+    step,
 }) => {
     const [parameterValue, setParameterValue] = useState<number>(value);
 
-    const floatingPointStep = maxValue === null ? DEFAULT_FLOAT_STEP : getFloatingPointStep(minValue, maxValue);
-
-    const step = type === 'int' ? DEFAULT_INT_STEP : floatingPointStep;
+    const fieldStep = getStep({ step, type, maxValue, minValue });
+    const formatOptions = type === 'float' ? { maximumFractionDigits: Math.abs(Math.log10(fieldStep)) } : undefined;
```
[the remark this answers is not in the document]
Comment (dwesolow) [reply web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/number-parameter-field.component.tsx:55]: AFAIK we don't have a number with more than 6 fraction digits in configuration.

**Review by jpggvilaca — APPROVED** [review]

File: web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/number-parameter-field.component.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (pplaskie) [reply web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/number-parameter-field.component.tsx:55]: But the LEARNING_RATE_STEP is 1e-6

[the remark this answers is not in the document]
Comment (pplaskie) [reply web_ui/src/pages/project-details/components/project-models/train-model-dialog/advanced-settings/ui/number-parameter-field.component.tsx:55]: Talked through offline

**Review by pplaskie — APPROVED** [review]
